services/user-management/src/main/java/com/foodie/user/contracts/RolePermitRequest.java

```diff
@@ -1,9 +1,11 @@
 package com.foodie.user.contracts;
 
+import com.foodie.user.model.Permission;
+
 import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 
 public record RolePermitRequest(
         @NotEmpty String id,
-        @NotNull PermissionRequest permission) {
+        @NotNull Permission permission) {
```

Comment:
What you had before is better

Reply:
Objects.NotNull is better suited for objects from what I read